src/main/java/org/broadinstitute/hellbender/tools/longreads/AlignedReadsToGraphConverter.java

```diff
@@ -0,0 +1,90 @@
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.argparser.ExperimentalFeature;
+import org.broadinstitute.hellbender.cmdline.programgroups.CoverageAnalysisProgramGroup;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.ReadWalker;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.tools.longreads.graph.AlignedBaseGraphCollection;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+@CommandLineProgramProperties(
+        summary = "Make a graph from a bam file containing aligned reads to a reference.  This tool will always create a .gfa file representing the graph of the aligned reads.",
+        oneLineSummary = "Make a graph from a bam file containing aligned reads to a reference.",
+        programGroup = CoverageAnalysisProgramGroup.class
+)
+@ExperimentalFeature
+public class AlignedReadsToGraphConverter extends ReadWalker {
+    private static final Logger logger = LogManager.getLogger(AlignedReadsToGraphConverter.class);
+
+    //==================================================================================================================
+    // Public Static Members:
+
+    //==================================================================================================================
+    // Private Static Members:
+
+    //==================================================================================================================
+    // Public Members:
+
+    @Argument(
+            fullName  = "output-file-base-name",
+            optional = true,
+            doc = "Base name for output files to be written.")
+    public String outputFileBaseName = "aligned_base_graph";
+
+    @Argument(
+            fullName  = "create-dot-files",
+            optional = true,
+            doc = "Create an additional DOT file for each GFA file created.")
+    public Boolean createDotFiles = false;
+
+    //==================================================================================================================
+    // Private Members:
+
+    private final AlignedBaseGraphCollection alignedBaseGraphCollection = new AlignedBaseGraphCollection();
+
+    //==================================================================================================================
+    // Constructors:
+
+    //==================================================================================================================
+    // Override Methods:
+
+    @Override
+    public void apply(final GATKRead read, final ReferenceContext referenceContext, final FeatureContext featureContext) {
+        // Add in our sequence data here:
+        alignedBaseGraphCollection.addSequence( read );
```

Comment:
I suspect it would be at this layer that you want to call MergeAdjacentNodesBefore_______. Because the graph itself doesn't know what order its seeing the sequences in or whether the input is in the correct order to even make that assumption. Furthermore its probably prudent to only call that code sufficiently upstream of where your processing is going on currently, so it might be worth keeping pool of reads and only calling that method every-so-often to ensure there is a balance between calling it infrequently and also calling it over already finished parts of the graph.